### PR TITLE
Disable shared lib by default

### PR DESCRIPTION
**Pull request by TeumessianFox**
QEMU meson detects all shared lib on the system which might be used by QEMU and links them. This is not a preferred behavior as it depends on the system which it is running on and its installed packages. As long as `libafl_qemu` does not detect all shared libs in its `build_linux.rs`, one needs to add them manually (https://github.com/AFLplusplus/LibAFL/blob/main/libafl_qemu/build_linux.rs#L299-L315). By disabling all features by default, this problem can be circumvented. If a specific feature is needed and present on a system, one can still enable it using the CLI flags `--enable-{feature}`.

I already tested this change and it does not break the libafl fuzzer `qemu_launcher`.

### CODE REVIEW

File: meson_options.txt [hunk 1]
```diff
@@ -86,212 +86,212 @@ option('cfi', type: 'boolean', value: 'false',
...
+option('tpm', type : 'feature', value : 'disabled',
        description: 'TPM support')
 
 # Do not enable it by default even for Mingw32, because it doesn't
 # work on Wine.
 option('membarrier', type: 'feature', value: 'disabled',
        description: 'membarrier system call (for Linux 4.14+ or Windows')
 
-option('avx2', type: 'feature', value: 'auto',
+option('avx2', type: 'feature', value: 'disabled',
```
Comment (andreafioraldi) [inline meson_options.txt:103]: why avx2 disabled?

Comment (TeumessianFox) [reply meson_options.txt:103]: I wasn't so sure about that one. It should be installed by default I think, but I couldn't confirm that it actually is.

Comment (TeumessianFox) [reply meson_options.txt]: I reactivated it as it seems to not be depended on a library (https://github.com/AFLplusplus/qemu-libafl-bridge/blob/main/meson.build#L2237-L2263) which we would need to link in libafl

**Comment by TeumessianFox** [comment]
Actually, I found a better way without changing `libafl_qemu_bridge`. Only some flags in `libafl_qemu` in `build_linux.rs` are needed.

**Pull request closed without merging**